GiTinder/Services/SettingsServices.cs

```diff
@@ -42,58 +44,82 @@ private bool SettingsExists(string username)
             {
-                Language foundLanguage = _context.Languages.Where(l => l.LanguageName == languageName).FirstOrDefault();
-                int languageId = foundLanguage.LanguageId;
+                int languageId = GetLanguageIdByLanguageName(languageName);
 
-                if (!_context.SettingsLanguage.Where(sl => sl.SettingsId == settingsId).Any(sl => sl.LanguageId == languageId))
+                if (!SettingsLanguageItemExistsInDb(settingsId, languageId))
                 {
                     AddSettingsLanguageItem(settingsId, languageId);
                 }
             }
+            List<string> LanguagesNamesListInLanguageDb = _languageServices.GetLanguageNamesListFromLanguageDb();
 
-            List<string> preferredLanguagesNamesListInDB = _context.SettingsLanguage.Select(sl => sl.Language.LanguageName).ToList();
-
-            foreach (string languageName in preferredLanguagesNamesListInDB)
+            foreach (string languageName in LanguagesNamesListInLanguageDb)
             {
-                Language foundLanguage = _context.Languages.Where(l => l.LanguageName == languageName).FirstOrDefault();
-                int languageId = foundLanguage.LanguageId;
+                int languageId = GetLanguageIdByLanguageName(languageName);
 
-                if (_context.SettingsLanguage.Where(sl => sl.SettingsId == settingsId).Any(sl => sl.LanguageId == languageId) &&
+                if (SettingsLanguageItemExistsInDb(settingsId, languageId) &&
                     !passedLanguageNameList.Contains(languageName))
                 {
                     RemoveSettingsLanguageItem(settingsId, languageId);
                 }
             }
         }
 
+        private Settings FindSettingsByUserToken(string usertoken)
+        {
+            int settingsId = FindSettingsIdByUsertoken(usertoken);
+            Settings settingsFoundInDb = FindSettingsBySettingsId(settingsId);
+            return settingsFoundInDb;
+        }
+
+        private int FindSettingsIdByUsertoken(string usertoken)
+        {
+            User foundUser = _userServices.FindUserByUserToken(usertoken);
+            int settingsId = foundUser.UserSettings.SettingsId;
+            return settingsId;
+        }
+
+        private Settings FindSettingsBySettingsId(int settingsId)
+        {
+            return _context.Settings.Where(s => s.SettingsId == settingsId).FirstOrDefault();
+        }
+
+        private bool SettingsLanguageItemExistsInDb(int settingsId, int languageId)
+        {
+            return _context.SettingsLanguage.Where(sl => sl.SettingsId == settingsId).Any(sl => sl.LanguageId == languageId);
+        }
+
+        private int GetLanguageIdByLanguageName(string languageName)
+        {
+            Language foundLanguage = _context.Languages.Where(l => l.LanguageName == languageName).FirstOrDefault();
+            int languageId = foundLanguage.LanguageId;
+            return languageId;
+        }
```

Comment:
@FilipTeply with these methods you still pretty much reinventing a wheel a little, but tomorrow I'll show what I have in mind